proposals/lip-0052.md

```diff
@@ -30,329 +31,345 @@ Therefore, providing a unified module to handle, transfer and modify NFTs is a n
-                      || uint32be(nftID.collection)
-                      || uint64be(nftID.index)
-
-        storeValue = {
-            "owner": address,
-            "lockingModuleID": NFT_NOT_LOCKED,
-            "attributes": givenAttributes
-        } serialized using NFTStoreSchema
-```
-
-#### deleteNFTEntry
-
-```python
-deleteNFTEntry(nftID):
-    delete the store entry with
-        storePrefix = STORE_PREFIX_NFT
-        storeKey = uint32be(nftID.chainID)
-                     || uint32be(nftID.collection)
-                     || uint64be(nftID.index)
-```
-
-#### createUserEntry
+If a state transition creates the `escrow` property of a non-existent substore entry, this entry is created following the `escrowStoreSchema`.
 
-```python
-createUserEntry(address, nftID):
-    create an store entry with
-        storePrefix = STORE_PREFIX_USER
-        storeKey = address
-                     || uint32be(nftID.chainID)
-                     || uint32be(nftID.collection)
-                     || uint64be(nftID.index)
-        storeValue = {"exists": True} serialized using userStoreSchema
-```
+#### Supported NFTs Substore
 
-#### deleteUserEntry
+- The substore prefix is set to `SUBSTORE_PREFIX_SUPPORTED_NFTS`.
+- Each store key is the `chainID` identifier of the chain to which the supported NFTs are native.
+- Each store value is the serialization of an object following `supportedNFTsStoreSchema`.
 
-```python
-deleteUserEntry(address, nftID):
-    delete the store entry with
-        storePrefix = STORE_PREFIX_USER
-        storeKey = address
-                     || uint32be(nftID.chainID)
-                     || uint32be(nftID.collection)
-                     || uint64be(nftID.index)
+```java
+supportedNFTsStoreSchema = {
+   "type": "object",
+   "required": ["supportedNftIDArray"],
+   "properties": {
+       "supportedNftIDArray" : {
+           "type": "array",
+           "fieldNumber": 1,
+           "items": {
+                "type": "object",
+                "required": ["nftID"],
+                "properties": {
+                    "nftID": {
+                        "dataType": "bytes",
+                        "length": NFT_ID_LENGTH,
+                        "fieldNumber": 1
+                    }
+                }
+            }
+       }
+   }
+}
 ```
 
-#### canonicalNFTID
+If all NFTs are supported, the substore contains an entry for the key ALL_SUPPORTED_NFTS_KEY and no other entries.
 
-```python
-canonicalNFTID(nftID):
-    if nftID.chainID == interoperability.getOwnChainAccount().ID
-        return {"chainID": CHAIN_ID_ALIAS_NATIVE, "collection": nftID.collection, "index": nftID.index}
-    else:
-        return nftID
-```
-
-This function will return the input `nftID` in the case `interoperabiliby.getOwnChainAccount()` cannot be called.
+If not all NFTs are supported, but all NFTs from a chain with `chainID` are supported, the substore contains an entry for key `chainID` with an empty array as value.
```

Comment:
So in my understanding if an entry exist for key `chainID` with a non-empty array as value, it means that we support only the specified NFTs from that chain? I am wondering if this level of granularity is really required and if we shouldn't just support or not NFTs from a certain chain. On the other hand, if we want something more granular, I would rather include the `collectionID` in the store value, so that we specify that we support all NFTs of a certain collection from a certain chain, by specifying a non-empty store value.

In all cases, I would add a line to explain this last example (entry existing with for key `chainID` with a non-empty array as value).

Reply:
This the supporting logic was copy-pasted from the Token LIP. I was wandering the same, namely if for NFTs it makes sense to support the individual NFTs instead of the whole collection. So, yes, I'm also in favor of supporting the whole collections instead of individual NFTs and I will change the logic accordingly.